### DISCUSSION TITLE
扩展编译平台不支持h5吗？

### DISCUSSION THREAD

**Discussion by thunur** [NervJS/taro#11935] [Q & A]
经过查阅文档自己写了个插件，在运行时编译器走的是小程序相关的，又找到个https://github.com/joyshion/taro-plugin-h5-platform-extend 这个通过入侵的方式扩展H5编译平台，但是在3.4.11时很多方法又用不了

**Comment by thunur** [comment]
难道是因为https://github.com/joyshion/taro-plugin-h5-platform-extend这个库的版本和主库的taro版本不一致导致的？

**Reply by biorz** [reply]
扩展编译平台好像是基于小程序模板编译那一套处理的， 不大清楚你这个需求点是什么，h5有什么不同需要单独拎出来处理。

**Reply by thunur** [reply]
> 扩展编译平台好像是基于小程序模板编译那一套处理的， 不大清楚你这个需求点是什么，h5有什么不同需要单独拎出来处理。

H5这边要跨端编译成云闪付的，

**Reply by Chen-jj** [reply]
@thunur 扩展编译平台目前都是针对类小程序平台的，可以大概描述一下你的需求

**Reply by thunur** [reply]
> @thunur 扩展编译平台目前都是针对类小程序平台的，可以大概描述一下你的需求

大佬你好，主要是这样，云闪付小程序的话它是类似于在app中用webview的方式加载html5界面，通过它提供的sdk文件进行和app交互，主要是想在现有的项目中，通过环境判断的方式加载云闪付小程序的sdk、dom渲染，如果能通过插件扩展编译平台的方式实现一个H5打包就可以满足这个需求了。例如process.env.TARO_ENV == 'yunshanfu' 这样能判断就很完美了~
我现在自己尝试了下使用ua的方式能实现，但是会有不需要的代码也会编译进来，导致体积变大。
按照现有的基础如果在增加一个扩展h5编译平台就更完美了。翼支付也是跟云闪付大体差不多的逻辑~
